### PR TITLE
feat: Migrate Palette to @artsy/palette-tokens package 

### PR DESCRIPTION
**Pull request by damassi**
This PR updates palette to point at palette-tokens, so that our atomic values can be shared with Eigen. Additionally, it updates palette-tokens with the latest v3 theme.

### CODE REVIEW

File: packages/palette-tokens/src/themes/v2.tsx [hunk 1]
```diff
@@ -4,8 +4,8 @@ import {
   TEXT_LETTER_SPACING,
   TEXT_LINE_HEIGHTS,
   TEXT_VARIANTS,
-} from "../elements/Text/tokens/v2"
-import { fontFamily } from "../platform/fonts"
+  fontFamily,
+} from "../typography/v2";
```
Comment (damassi) [inline packages/palette-tokens/src/themes/v2.tsx:8]: not sure whats up with these semis

**Comment by damassi** [comment]
@dzucconi - need to give publishing a look tomorrow but assigning to you for review in the meantime.

File: packages/palette/src/Theme.tsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ import { GridThemeProvider as StyledGridThemeProvider } from "styled-bootstrap-g
 import { ThemeContext, ThemeProvider } from "styled-components"
 import { Theme as TTheme, THEME_V2, THEME_V3, ThemeV2, ThemeV3 } from "./themes"
 
-export * from "./themes/v2"
+export * from "@artsy/palette-tokens/dist/themes/v2"
```
Comment (dzucconi) [inline packages/palette/src/Theme.tsx:9]: Wondering if you're going to have to just manually list out the specific exports as these are failing in type-check

**Review by dzucconi — APPROVED** [review]
All looks reasonable to me

File: packages/palette/src/Theme.tsx (hunk 2 above)
Comment (damassi) [reply packages/palette/src/Theme.tsx:9]: Type-checking passes locally, but for some reason it's blowing up in CI. I'm not sure why. I even pointed things at canaries.

**Comment by damassi** [comment]
Merging, as CI seems to be referencing an old `node_modules` cache. Verified by SSHing into box and running `yarn add @artsy/palette-tokens` and seeing job pass, yet after the addition there was no change to the lock file indicating CI is referencing an old cache key.

**Comment by artsyit** [comment]
:rocket: PR was released in [`@artsy/palette-tokens@2.1.0`](https://github.com/artsy/palette/releases/tag/%40artsy/palette-tokens%402.1.0), [`@artsy/palette@14.44.0`](https://github.com/artsy/palette/releases/tag/%40artsy/palette%4014.44.0) :rocket:

**Pull request merged**